allel/stats/roh.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, division
+
+import numpy as np
+from hmmlearn import hmm
+from allel.chunked import ChunkedArray
+
+
+def call_roh(g, pos, ix, emission=(0.001, 0.0025, 0.01),
+             transition=1e-6, min_roh=100000,
+             is_accessible=None, contig_size=None):
+
+    """
+    Call ROH (runs of homozygosity) of a given individual
+
+    Parameters
+    ----------
+    g : array_like, int, shape (n_variants, n_samples, ploidy)
+        Genotype array.
+    pos: array_like, int, shape (n_variants)
+        Positions of variants, same 0th dimension as g
+    ix: integer
+        index of the sample on which to call roh
+    emission: tuple or array_like, float
+        Probability of observing a heterozygote in each hidden state.
+        Supports >= 2 states. Appropriate values will depend on
+        frequency of sequencing error & spurious het calls due
+        to misalignment
```

Comment:
To help the user, can we split this into two arguments? 

E.g.,. first argument "p_het_roh" is the probability of observing a heterozygous genotype within a run of homozygosity. This is essentially the genotyping error probability.

E.g., second argument "p_het" is the (non-ROH) heterozygosity, i.e., nucleotide diversity. Here the user can provide a single value, being the average heterozygosity for the species/population, or could provide multiple values if you want to model different levels of non-ROH heterozygosity in the genome, as we've done.